packages/tpetra/core/src/Tpetra_Distributor.hpp

```diff
@@ -2823,6 +2866,7 @@ namespace Tpetra {
           size_t j = startsTo_[p];
           size_t numPacketsTo_p = 0;
           for (size_t k = 0; k < lengthsTo_[p]; ++k, ++j) {
+            numPacketsTo_p += numExportPacketsPerLID[j];
```

Comment:
Note to self: This is the line that fixes the bug :-) .

Reply:
(The changes to other lines are important because they help with debug output, but they do not actually fix the issue.)